examples/confetti_snapshot_animation/animated_explosion/imp.rs

```diff
@@ -0,0 +1,49 @@
+use crate::Explosion;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{gdk, glib};
+use std::cell::{Cell, RefCell};
+
+#[derive(Default)]
+pub struct AnimatedExplosion {
+    pub(super) explosion: RefCell<Option<Explosion>>,
+    pub(super) start: Cell<f64>,
+    pub(super) lastupdate: Cell<f64>,
+    pub(super) duration: Cell<f64>,
+    pub(super) finished: Cell<bool>,
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for AnimatedExplosion {
+    const NAME: &'static str = "AnimatedExplosion";
+    type Type = super::AnimatedExplosion;
+}
+
+impl ObjectImpl for AnimatedExplosion {}
+
+impl AnimatedExplosion {
+    pub(super) fn init_time(&self, clock: &gdk::FrameClock, duration: f64) {
+        let time = clock.frame_time() as f64 / 1000.0;
+        self.start.replace(time);
+        self.lastupdate.replace(time);
+        self.duration.replace(duration);
+    }
+    // Returns ControlFlow::Break if the animation is finished
+    pub(super) fn update(&self, clock: &gdk::FrameClock) -> Continue {
```

Comment:
As much as I really like the usage of ControlFlow here, I think we should wait until we do the switch in glib-rs first & the generated bindings.